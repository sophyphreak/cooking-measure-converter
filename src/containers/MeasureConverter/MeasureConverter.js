import React from 'react';
import ConverterComponent from '../../components/ConverterComponent/ConverterComponent';
import doConversion from './doConversion/doConversion';
import isInvalidNumber from './validation/isInvalidNumber';
import massConverter from './converters/massConverter';

// TODO
//
// Convert this.state into series of objects
// each object has (1) imperialX (2) metricX (3) imperialUnit (4) metricUnit
// When unit changes, it changes the OTHER number

export default class MeasureConverter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      mass: {
        imperialMass: '',
        imperialUnit: 'lb',
        metricMass: '',
        metricUnit: 'kg'
      }
    };
    this.onImperialMassChange = this.onImperialMassChange.bind(this);
    this.onMetricMassChange = this.onMetricMassChange.bind(this);
  }

  onImperialMassChange(e) {
    const imperialMass = e.target.value;
    if (isInvalidNumber(imperialMass)) {
      return;
    }
    const { imperialUnit, metricUnit } = this.state.mass;
    const conversionInputs = {
      inputAmount: imperialMass,
      inputUnit: imperialUnit,
      outputUnit: metricUnit
    };
    const metricMass = doConversion(conversionInputs);
    const mass = {
      imperialMass,
      imperialUnit,
      metricMass,
      metricUnit
    };
    this.setState(() => ({ mass }));
  }

  onMetricMassChange(e) {
    const metricMass = e.target.value;
    const converterInputs = {
      direction: 'metricToImperial',
      inputAmount: metricMass,
      inputMassState: this.state.mass
    };
    const mass = massConverter(converterInputs);
    this.setState(() => ({ mass }));
  }

  render() {
    const massProps = {
      mass: this.state.mass,
      onImperialMassChange: this.onImperialMassChange,
      onMetricMassChange: this.onMetricMassChange
    };
    return <ConverterComponent massProps={massProps} />;
  }
}
