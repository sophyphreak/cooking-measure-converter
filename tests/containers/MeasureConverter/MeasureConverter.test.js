import React from 'react';
import { shallow } from 'enzyme';
import MeasureConverter from '../../../src/containers/MeasureConverter/MeasureConverter';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<MeasureConverter />);
});

test('should render MeasureConverter correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
