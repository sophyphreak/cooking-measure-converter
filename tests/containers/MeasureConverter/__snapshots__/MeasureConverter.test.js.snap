// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MeasureConverter correctly 1`] = `
<ConverterComponent
  lengthProps={
    Object {
      "lengthState": Object {
        "imperialUnit": "in",
        "metricUnit": "cm",
      },
      "lengthUnitOptions": Object {
        "imperialUnitOptions": Array [
          "ft",
          "in",
        ],
        "metricUnitOptions": Array [
          "m",
          "cm",
          "mm",
        ],
      },
      "onImperialLengthChange": [Function],
      "onMetricLengthChange": [Function],
    }
  }
  massProps={
    Object {
      "massState": Object {
        "imperialUnit": "lb",
        "metricUnit": "kg",
      },
      "massUnitOptions": Object {
        "imperialUnitOptions": Array [
          "lb",
          "oz",
        ],
        "metricUnitOptions": Array [
          "kg",
          "g",
        ],
      },
      "onImperialMassChange": [Function],
      "onMetricMassChange": [Function],
    }
  }
  temperatureProps={
    Object {
      "onImperialTemperatureChange": [Function],
      "onMetricTemperatureChange": [Function],
      "temperatureState": Object {
        "imperialTemperature": "",
        "metricTemperature": "",
      },
    }
  }
  volumeProps={
    Object {
      "onImperialVolumeChange": [Function],
      "onMetricVolumeChange": [Function],
      "volumeState": Object {
        "imperialUnit": "Tbs",
        "metricUnit": "ml",
      },
      "volumeUnitOptions": Object {
        "imperialUnitOptions": Array [
          "gal",
          "qt",
          "pnt",
          "cup",
          "Tbs",
          "tsp",
        ],
        "metricUnitOptions": Array [
          "l",
          "ml",
        ],
      },
    }
  }
/>
`;
